File: platforms/react-native/src/distance.ts
/**
 * Nautical-mile ↔ meter conversion.
 */

export const Distance = {
  /** Meters in one nautical mile. */
  metersPerNauticalMile: 1852,
} as const;

/** Interprets `nm` as nautical miles and returns the equivalent in meters. */
export function nauticalMilesToMeters(nm: number): number {
  return nm * Distance.metersPerNauticalMile;
}

/** Interprets `meters` as meters and returns the equivalent in nautical miles. */
export function metersToNauticalMiles(meters: number): number {
  return meters / Distance.metersPerNauticalMile;
}

File: packages/unity/com.zibaltech.geonavkit/Runtime/Distance.cs
//
//  Distance.cs
//  GeoNavKit
//
//  Nautical-mile ↔ meter conversion.
//

namespace GeoNavKit
{
    public static class Distance
    {
        /// <summary>Meters in one nautical mile.</summary>
        public const double MetersPerNauticalMile = 1852.0;

        /// <summary>Interprets <paramref name="nauticalMiles"/> as NM and returns meters.</summary>
        public static double NauticalMilesToMeters(this double nauticalMiles) =>
            nauticalMiles * MetersPerNauticalMile;

        /// <summary>Interprets <paramref name="meters"/> as meters and returns nautical miles.</summary>
        public static double MetersToNauticalMiles(this double meters) =>
            meters / MetersPerNauticalMile;
    }
}
